packages/story-editor/src/app/pageCanvas/test/usePageCanvas.js

```diff
@@ -0,0 +1,270 @@
+          await usePageCanvasCurrent.actions.calculateAccessibleTextColors(
+            element1
+          );
+        });
+        expect(storyPageToCanvas).toHaveBeenCalledTimes(1);
+
+        // cache already populated with generated canvas for currentPage
+        await act(async () => {
+          const [usePageCanvasCurrent] = result.current;
+          await usePageCanvasCurrent.actions.calculateAccessibleTextColors(
+            element2
+          );
+        });
+        expect(storyPageToCanvas).toHaveBeenCalledTimes(1);
+      });
+
+      it('generates a page canvas when called and canvas cache is stale', async () => {
+        const { result } = renderUsePageCanvas(mockStoryContext);
+        const [, useStoryMockCurrent] = result.current;
+        const { actions: storyActions } = useStoryMockCurrent;
+
+        const element1 = { id: 'a' };
+        const element2 = { id: 'b' };
+
+        // cache populates with generated canvas for currentPage
+        await act(async () => {
+          const [usePageCanvasCurrent] = result.current;
+          await usePageCanvasCurrent.actions.calculateAccessibleTextColors(
+            element1
+          );
+        });
+        expect(storyPageToCanvas).toHaveBeenCalledTimes(1);
+
+        // make currentPage stale by making an update to it
+        act(() => storyActions.duplicateLastElement());
+        expect(result.current[0].state.pageCanvasMap).toStrictEqual({});
+
+        // cache populates with generated canvas for updated currentPage
+        await act(async () => {
+          const [usePageCanvasCurrent] = result.current;
+          await usePageCanvasCurrent.actions.calculateAccessibleTextColors(
+            element2
+          );
+        });
+        expect(storyPageToCanvas).toHaveBeenCalledTimes(2);
+      });
+    });
+
+    describe('partial page snapshot cache', () => {
+      it('generates a partial page canvas when called and partial page snapshot cache is not populated', async () => {
+        // set up story to have single selected element
+        const lastElementIndex = currentPage.elements.length - 1;
+        const selectedElement = currentPage.elements[lastElementIndex];
+        mockStoryContext.state.selectedElementIds = [selectedElement.id];
+
+        // render hook
+        const { result } = renderUsePageCanvas(mockStoryContext);
+
+        // since we're testing an existing element against the currentPage
+        // calculateAccessibleTextColors knows to exclude the element we're testing
+        // from the generated canvas. It holds this canvas generated from a partial
+        // of the page in the snapshot cache.
+        await act(async () => {
+          const [usePageCanvasCurrent] = result.current;
+          await usePageCanvasCurrent.actions.calculateAccessibleTextColors(
+            selectedElement
+          );
+        });
+        expect(storyPageToCanvas).toHaveBeenCalledTimes(1);
```

Comment:
🤔 I guess it's not possible to test that a partial snapshot vs a whole snapshot was created since both use the same `storyPageToCanvas` fn, right?